### PR TITLE
Made modifications to the GUI.

### PR DESCRIPTION
**Pull request by Wang-yuanlu**
Hi, Mr Tan:
I recently modified the GUI to make it adapt to the window size. However, the "CGRAPannel" located in the top left corner still only adapts to a size of 4x4. I also added scrollbars, but the results were not satisfactory. The rest of the GUI is mostly fine without any apparent issues. I will think of another solution to address the problem in the top left corner.
Thank you!
Have a nice day!

### CODE REVIEW

**Comment by Wang-yuanlu** [comment]
I realized only when submitting the pull request that the changes I made locally after forking would be included in the submission. Unfortunately, I ignored the details and wrote some comments in Chinese. It seems that those commits may not meet the required standards.

**Comment by tancheng** [comment]
Hi Yuanlu, thanks for the contribution, there is no specific standard. Any contribution is appreciated.
For this PR, can you summarize the modifications/features briefly here please? I saw you also resolve some DISPLAY issues. You can also attach two photos to show how the window size would affect the sub-components' position & size. Thanks!

**Comment by Wang-yuanlu** [comment]
Hi, Mr Tan:
 I recently finished refining the GUI. Here are the modifications I made:
1. I divided the entire GUI window into a grid of 6 rows and 6 columns to accommodate six modules: cgrapannel, kernelpannel, mappingpannel, layoutpannel, parampannel, and functionpannel (including testpannel, verilogpannel, and reportpannel sections).
2.  Instead of using "tkinter. Button" to create the Tiles, I used "canvas.create_rectangle" to facilitate the implementation of adaptive behavior in conjunction with the "tkinter. Scrollbar" method for the scrollbar.
3. To address the issue of the report module overflowing, I modified the layout from three rows and four columns to six rows and two columns. I also redistributed the available space to prevent overflowing to the right.
4. During the modification process, there was an issue with the "clickShowDFG" button not working. By utilizing the Python debugger and setting breakpoints in the code, I discovered that a trace statement was missing, which resulted in abnormal generation of the "kernel.bc" file. Now the problem has been fixed.
5. If you need to adjust the size of the model, please modify the length and width parameters in lines 29-33 of the `launchUI.py` file, as well as the model's row and column parameters in lines 49 and 50.
GUI bigger
GUI smaller

**Comment by Wang-yuanlu** [comment]
Hi Mr Tan, I have already submitted my modifications and attached a schematic diagram demonstrating the visual effects. I hope that these modifications meet the project requirements.
Have a nice day!


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "tancheng/CGRA-Flow"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年4月27日(星期六) 晚上11:42
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [tancheng/CGRA-Flow] Made modifications to the GUI. (PR #26)





 
Hi Yuanlu, thanks for the contribution, there is no specific standard. Any contribution is appreciated.
 For this PR, can you summarize the modifications/features briefly here please? I saw you also resolve some DISPLAY issues. You can also attach two photos to show how the window size would affect the sub-components' position &amp; size. Thanks!

**Comment by tancheng** [comment]
Hi Yuanlu, thanks for the summary.

Using the `create_rectangle` instead of `tkinter. Button` for tile causes some issues when select the functional units for the chosen tile, right? Allowing the tile to be selectable is more important. Can we fix this then we can submit this PR afterwards?

Can you briefly explain why the DISPLAY port 12 helps? And why should we use `--name=CGRA-Flow-v2` or we it needs to be changed back to `--name=CGRA-Flow`?

**Comment by Wang-yuanlu** [comment]
Hi, Mr.Tan, I have corrected the issue with "cgrapannel" and reverted to using the Button method. The modification for DISPLAY port 12 was a separate change I made while configuring the environment on the laboratory server. I have now reverted it to its original form.
I further optimized the GUI display size, and although there may still be some aesthetic flaws, it is now functional. I will gradually work on improving the size of the paramPannel and cgraPannel in the future to make the entire interface more visually appealing.
Thanks for your help.
Have a nice day!
bigger
smaller

**Comment by tancheng** [comment]
Thanks Yuanlu, can you try to adjust the entire window size to fit/match the 4x4 CGRA design, and set that size as **default** entire window size when the UI is launched?

**Comment by Wang-yuanlu** [comment]
Hi, Mr.Tan. I have resized the entire window to fit a 4x4 CGRA disign and set an appropriate initial window size.

**Comment by Wang-yuanlu** [comment]
Hi, Mr.Tan. I have resized the entire window to fit a 4x4 CGRA disign and set an appropriate initial window size.
